src/braket/circuits/observables.py

```diff
@@ -318,9 +356,9 @@ def _get_eigendecomposition(self) -> Dict[str, np.ndarray]:
             corresponding eigenvectors in the "eigenvectors" matrix. These cached values
             are immutable.
         """
-        mat_key = tuple(self._matrix.flatten().tolist())
+        mat_key = tuple(matrix.flatten().tolist())
         if mat_key not in Hermitian._eigenpairs:
-            eigenvalues, eigenvectors = np.linalg.eigh(self._matrix)
+            eigenvalues, eigenvectors = np.linalg.eigh(matrix)
             eigenvalues.setflags(write=False)
             eigenvectors_conj_t = eigenvectors.conj().T
```

Comment:
Personally I'd probably just store eigenvectors instead of eigenvectors_conj_t for better readability. No big deal though.

Reply:
I did this initially, but opted for existing way for a _hair_ better performance (so we don't have to take the conjugate transpose every time a cached matrix is instantiated). I was still on the fence, and this comment was the push I needed.